lib/vfscore/README.md

```diff
@@ -0,0 +1,131 @@
+There are multiple types of filesystems available in Unikraft (e.g: `ramfs`,` 9pfs`).
+
+## Key Functions and Data Structures
+
+The `vfscore` library provides 2 structures to define operations on a filesystem (defined in [lib/vfscore/include/vfscore/mount.h](/lib/vfscore/include/vfscore/mount.h)):
+
+```c
+struct vfsops {
+        int (*vfs_mount)        (struct mount *, const char *, int, const void*);
+        ...
+        struct vnops    *vfs_vnops;
+};
+```
+
+```c
+struct vnops {
+        vnop_open_t             vop_open;
+        vnop_close_t            vop_close;
+        vnop_read_t             vop_read;
+        vnop_write_t            vop_write;
+        vnop_seek_t             vop_seek;
+        vnop_ioctl_t            vop_ioctl;
+        ...
+};
+```
+
+The first structure mainly defines the operation of mounting the filesystem, while the second defines the operations that can be executed on files (regular files, directories, etc).
+The `vnops` structure can be seen as the `file_operation` structure in the Linux Kernel (more as an idea).
+More about this structure [here](https://tldp.org/LDP/lkmpg/2.4/html/c577.htm).
+
+The filesystem library will define two such structures through which it will provide the specified operations.
+To understand how these operations end up being used let's examine the open system call:
+
+```c
+int
+sys_open(char *path, int flags, mode_t mode, struct vfscore_file **fpp)
+{
+        struct vfscore_file *fp;
+        struct vnode *vp;
+        ...
+        error = VOP_OPEN(vp, fp);
+}
+```
+
+`VOP_OPEN()` is a macro that is defined as follows:
+
+```c
+#define VOP_OPEN(VP, FP)           ((VP)->v_op->vop_open)(FP)
+```
+
+## Register a new filesystem
+
+
+To map the desired function for your filesystem with the vfscore application, you need to have declare a `struct vnops` and populate the members, as shown below:
+
+```c
+struct vnops ramfs_vnops = {
+		ramfs_open,             /* open */
+		ramfs_close,            /* close */
+		ramfs_read,             /* read */
+		ramfs_write,            /* write */
+		ramfs_seek,             /* seek */
+		ramfs_ioctl,            /* ioctl */
+		ramfs_fsync,            /* fsync */
+		ramfs_readdir,          /* readdir */
+		ramfs_lookup,           /* lookup */
+		ramfs_create,           /* create */
+		ramfs_remove,           /* remove */
+		ramfs_rename,           /* remame */
+		ramfs_mkdir,            /* mkdir */
+		ramfs_rmdir,            /* rmdir */
+		ramfs_getattr,          /* getattr */
+		ramfs_setattr,          /* setattr */
+		ramfs_inactive,         /* inactive */
+		ramfs_truncate,         /* truncate */
+		ramfs_link,             /* link */
+		(vnop_cache_t) NULL,    /* arc */
+		ramfs_fallocate,        /* fallocate */
+		ramfs_readlink,         /* read link */
+		ramfs_symlink,          /* symbolic link */
+		ramfs_poll,             /* poll */
+};
```

Comment:
I wouldn't add the entire structure here, do some `[...]` in the middle.
Also add a link to where these are all defined, so the user can take a look at it more in depth.